torch/custom_class.h

```diff
@@ -101,9 +102,23 @@ class class_ {
     defineMethod<void>("__init__", std::move(func));
     return *this;
   }
-  template <typename Func>
+  template <
+      typename Method,
+      std::enable_if_t<
+          std::is_member_function_pointer<std::decay_t<Method>>::value,
+          bool> = false>
+  class_& def(std::string name, Method m) {
+    auto res =
+        def_(std::move(name), std::move(m), detail::args_t<decltype(m)>{});
+    return *this;
+  }
+  template <
+      typename Func,
+      std::enable_if_t<
+          !std::is_member_function_pointer<std::decay_t<Func>>::value,
+          bool> = false>
   class_& def(std::string name, Func f) {
```

Comment:
You should probably use perfect forwarding instead of moving for `f` so that callers can pass in const references if they want to. Sorry, I might have given wrong advice here earlier.